R/ebird_import.R

```diff
@@ -0,0 +1,283 @@
+#------------------------------------------------------------------------------#
+# Functions/workflows for loading ebird data from file(s)
+#------------------------------------------------------------------------------#
+
+#' Create a data.frame listing available eBird files
+#' @include config.R
+list_ebird_files <- function() {
+  list.files(config$ebird_data_dir) |>
+    (\(x) {
+      dplyr::tibble(
+        file   = x,
+        county = stringr::str_extract(x, "Durham|Orange|Chatham"),
+        export_date = stringr::str_extract(x, "\\d{8}")
+      )
+    })() |>
+    dplyr::group_by(county) |>
+    dplyr::mutate(
+      latest = export_date == max(export_date)
+    )
+}
+
+#' Column specification for ebird data
+#' @importFrom readr cols col_character col_skip col_number col_integer col_date
+#'                   col_time
+ebird_cols <- cols(
+  `Submission ID`          = col_character(),
+  `Common Name`            = col_character(),
+  `Scientific Name`        = col_character(),
+  `Taxonomic Order`        = col_skip(),
+  Count                    = col_character(),
+  `State/Province`         = col_skip(),
+  County                   = col_skip(),
+  `Location ID`            = col_skip(),
+  Location                 = col_character(),
+  Latitude                 = col_number(),
+  Longitude                = col_number(),
+  Date                     = col_date(format = "%Y-%m-%d"),
+  Time                     = col_time(),
+  Protocol                 = col_skip(),
+  `Duration (Min)`         = col_skip(),
+  `All Obs Reported`       = col_skip(),
+  `Distance Traveled (km)` = col_skip(),
+  `Area Covered (ha)`      = col_skip(),
+  `Number of Observers`    = col_integer(),
+  `Breeding Code`          = col_skip(),
+  `Observation Details`    = col_character(),
+  `Checklist Comments`     = col_character(),
+  `ML Catalog Numbers`     = col_skip()
+)
+
+#' Load the *most recent* ebird export per county
+#' (i.e. MBBS ebird account)
+#' into an R `data.frame`.
+#' according the `ebird_cols` specification.
+#' Additionally renames the variables.
+#'
+#' NOTE: readr::read_csv gives warnings upon parsing the ebird CSVs.
```

Comment:
Create issue regarding this note.

Reply:
Yeah, for all the csvs. You can see in the first non-header row of (e.g. `MyEBirdData_Chatham_20240730.csv`): there's no comma after the checklist comment, but -- per the header -- there should be one for `ML Catalog Numbers`.

This reminds me though, we need to add a note somewhere to download ebird files directly and **not open with excel**.